### ISSUE TITLE
Offline GOG JSON installers with version: GOG and gogid: fail with "Token required"

### ISSUE THREAD

**Issue by jmczab** [lutris/lutris#5782]
### Bug description

Version: 5.18 Git version, updated 2024-12-10
When using a Downloaded or clicked GoG "offline" installer - e.g. [system shock 2](https://lutris.net/games/system-shock-2) WINE GoG version, you get a "No Token Available" error, indicating it requires the GoG source to be enabled and logged in.

The JSON (and install link) includes "version:" "GOG" and "gogid:" "123458"  which triggers the fetching of data from GoG services.

If "version:" is changed, or gogid: removed from a downloaded installer, then the error no longer occours.

Note that using the YAML source (middle tab) does not trigger the error as it has no gogid.

### How to Reproduce

Steps to reproduce the behaviour:
1. Open the Lutris client
2. Disconnect (and disable) the GoG source
3. Download any GoG game there is an offline installer for (picked System Shock 2 for the small download)
4. Go to Lutris.net, find the game, click the install link, or click download and run the script
5. Attempt to install, after clicking next on the destination directory, experience the Token error
6. Close Lutris
7. Download the script and remove the line with gogid: or change version: to something random
8. Re-run install

### Expected behavior

The installer needs to be able to handle truly offline installs.
What conditions need to be satisfied for not connecting? Maybe if the file is locally selected (not downloaded?) and no GoG source is enabled.

Only titles installed via the source using an Auto install should need to fetch data. Everything else should/could technically be offline.

### Log output

```shell
No GOG token available

Traceback (most recent call last):

  File "/home/shodan/source/lutris/lutris/util/jobs.py", line 30, in target
    result = self.function(*a, **kw)
             ^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/shodan/source/lutris/lutris/installer/interpreter.py", line 204, in get_extras
    return self.service.get_extras(self.installer.service_appid)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/shodan/source/lutris/lutris/services/gog.py", line 341, in get_extras
    game = self.get_game_details(gogid)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/shodan/source/lutris/lutris/services/gog.py", line 291, in get_game_details
    return self.make_api_request(url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/shodan/source/lutris/lutris/services/gog.py", line 228, in make_api_request
    token = self.load_token()
            ^^^^^^^^^^^^^^^^^

  File "/home/shodan/source/lutris/lutris/services/gog.py", line 202, in load_token
    raise AuthenticationError("No GOG token available")

lutris.exceptions.AuthenticationError: No GOG token available

Lutris log:
[INFO:2024-12-10 11:47:09,611:application]: Starting Lutris 0.5.18
[INFO:2024-12-10 11:47:09,663:startup]: "card1" is AMD Radeon RX 7900 XT (1002:744c 1043:05ed amdgpu) Driver 24.0.9
[DEBUG:2024-12-10 11:47:09,828:xrandr]: Retrieving resolution list
[DEBUG:2024-12-10 11:47:09,842:xrandr]: Retrieving 25 video modes from XrandR
[DEBUG:2024-12-10 11:47:09,930:path_cache]: Checking for missing games
[WARNING:2024-12-10 11:47:09,967:base]: Tried to get auth status while login in progress
[DEBUG:2024-12-10 11:47:09,978:api]: Modified at Tue 10 Dec 2024 09:25:43 , will update after Tue 10 Dec 2024 15:25:43 
[DEBUG:2024-12-10 11:47:09,979:lutriswindow]: Runtime up to date
[WARNING:2024-12-10 11:47:29,000:battlenet]: The Battle.net source is unavailable because Google protobuf could not be loaded: No module named 'google'
[INFO:2024-12-10 11:47:29,164:application]: Starting Lutris 0.5.18
[INFO:2024-12-10 11:47:29,215:startup]: "card1" is AMD Radeon RX 7900 XT (1002:744c 1043:05ed amdgpu) Driver 24.0.9
[DEBUG:2024-12-10 11:47:30,901:__init__]: Action set to play
[DEBUG:2024-12-10 11:47:30,901:__init__]: Action set to install
[DEBUG:2024-12-10 11:47:31,859:api]: Fetching installer https://lutris.net/api/installers/system-shock-2-gog
[DEBUG:2024-12-10 11:47:31,860:http]: GET https://lutris.net/api/installers/system-shock-2-gog
[DEBUG:2024-12-10 11:47:32,036:application]: Showing window InstallerWindowsystem-shock-2
[DEBUG:2024-12-10 11:47:35,221:xrandr]: Retrieving 25 video modes from XrandR
[DEBUG:2024-12-10 11:47:57,472:xrandr]: Retrieving 25 video modes from XrandR
[DEBUG:2024-12-10 11:48:04,883:gog]: Download extras for GOG ID 1207659172 and its DLCs
[INFO:2024-12-10 11:48:04,884:gog]: Getting game details for 1207659172
[ERROR:2024-12-10 11:48:04,884:jobs]: Error while completing task <bound method ScriptInterpreter.get_extras of <interpreter.ScriptInterpreter object at 0x72ed56fba080 (lutris+installer+interpreter+ScriptInterpreter at 0x3d17b60)>>: <class 'lutris.exceptions.AuthenticationError'> No GOG token available
```

### System Information

```shell
Vulkan support: YES
Esync support: YES
Fsync support: YES
Wine installed: YES
Gamescope: NO
Mangohud: NO
Gamemode: YES
Steam: YES
In Flatpak: NO
[System]
OS: Linux Mint 22 Wilma
Arch: x86_64
Kernel: 6.8.0-49-generic
Lutris Version: 0.5.18
Desktop: X-Cinnamon
Display Server: x11
[CPU]
Vendor: AuthenticAMD
Model: AMD Ryzen 9 3900X 12-Core Processor
Physical cores: 12
Logical cores: 24
[Memory]
RAM: 31.2 GB
Swap: 1.9 GB
[Graphics]
Vendor: AMD
OpenGL Renderer: AMD Radeon RX 7900 XT (radeonsi, navi31, LLVM 17.0.6, DRM 3.57, 6.8.0-49-generic)
OpenGL Version: 4.6 (Compatibility Profile) Mesa 24.0.9-0ubuntu0.2
OpenGL Core: 4.6 (Core Profile) Mesa 24.0.9-0ubuntu0.2
OpenGL ES: OpenGL ES 3.2 Mesa 24.0.9-0ubuntu0.2
Vulkan Version: 1.3.275
Vulkan Drivers: AMD Radeon RX 7900 XT (RADV NAVI31) (1.3.274)
```

**Comment by legluondunet** [comment]
I don't understand, if you connect your GOG account to Lutris, you have choice to download OR select an offline installer.
So why you don't want to connect your GOG account? It is the way to follow to avoid your issue.

**Comment by jmczab** [comment]
> I don't understand, if you connect your GOG account to Lutris, you have choice to download OR select an offline installer. So why you don't want to connect your GOG account to Lutris? It is the way to follow to avoid your issue.

(Custom) Installers authored before the GoG source was added did not error, so this is a change in Lutris behaviour which breaks all if the source is not enabled.

Those installers should continue to install regardless of whether GoG exists or not - something that has been a major selling point for them recently :)

**Comment by PerAstraAdDeum** [comment]
I am unable to use the offline installer for any Game I purchased of and downloaded from GOG.

Lutris always tells me "No GOG token available". I do not wish to link my GOG account with Lutris, nor was that ever a requirement. I assume this is a bug.

**Comment by NightsPaladin** [comment]
Chiming in with a +1 to this. I believe there are use cases where connecting their GOG account to Lutris isn't something people will want to be forced to do, after that has not been the case for years.

**Comment by danieljohnson2** [comment]
Okay, okay, I'm awake again.

This seems to be a pretty old issue.

It is because we've made the exception handling more specific, so it will let you install if GOG doesn't have the game anymore but not for any other error. It needs to handle ``AuthenticationError``.

It's also because we can download extras (like soundtracks) from GOG, but if this fails for any reason you just can't install..

I've improved the error handling, and I can install System Shock 2 with the custom script now.

**Comment by PerAstraAdDeum** [comment]
Please stop nudging people into linking their accounts to GOG. I bought my games from GOG because they sell DRM-free products and make offline-installer available for every game.
If I wanted to have a launcher for downloading, installing and running my games, I'd just have gone with Steam.

**Comment by danieljohnson2** [comment]
Are you asking me to revert the fix so the bug stays in? Or do you just not want to use Lutris, a launcher that downloads, installs, and runs you games?

:confused:

**Comment by jmczab** [comment]
Well, thank you - I'm happy the installers I have offline once again work without me having to link up GoG on my old 2nd kitchen PC that I frequently rebuild (whilst I bake, usually) to test the latest python/distro/WINE/drivers.
I'm happier with the ease at which I can use tools like Lutris to install my partly offline GoG library :+1:

**Comment by legluondunet** [comment]
I'm not sure this resolved issue is a good thing, apart for piracy people...

**Comment by danieljohnson2** [comment]
I wouldn't agree with that. Bear in mind that this issue did not ever prevent you from just installing a GoG offline installer - you couldn't use certain scripts to do so, like the System Shock 2 script mentioned by the OP. Just installing the thing itself worked fine.

That's probably why this has been outstanding for so long, without complaints until this one.

**Comment by legluondunet** [comment]
I understand better, thank you for this explanation

**Comment by meedstrom** [comment]
In case anyone's simply wondering *how* to link their GOG account.  I searched for the longest time...  The error message doesn't hold your hand at all.

TL;DR: it's not in the settings menu but the main Lutris window, sidebar, Sources section, click on the person-figure next to GOG.
